### PR TITLE
Implement crash reporting permission ask dialog

### PR DESCRIPTION
**Pull request by simonhong**
Launch crash reporting permission ask dialog on startup after crash.
Launch tab restore dialog after permission ask dialog is closed.

By overriding, ShowIfNotOffTheRecordProfileBrave() is called instead of
ShowIfNotOffTheRecordProfile().
ShowIfNotOffTheRecordProfileBrave() will launch crash reporting permission
ask dialog and then tab restore dialog is launched after permission ask dialog
is closed.

Screen Shot 2021-07-27 at 9 56 19 PM
Screen Shot 2021-07-27 at 9 56 38 PM

Resolves https://github.com/brave/brave-browser/issues/1841

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`npm run test brave_unit_tests -- --filter=MetricsUtilTest*`
`npm run test brave_browser_tests -- --filter=CrashReportPermissionAskDialogTest*`

1. Launch browser and open several tabs
2. Check crash report is not enabled from brave://settings/privacy
3. Load brave://inducebrowsercrashforrealz to simulate browser crash
4. Check crash reporting permission ask dialog is launched
4-1. Click Block and check tab restore bubble is displayed
4-2. Click Allow and check tab restore bubble is displayed
4-2-1. Check crash reporting is enabled from brave://settings/privacy

### CODE REVIEW

**Comment by simonhong** [comment]
will push test code soon.

**Review by mariospr — APPROVED** [review]
LGTM for chromium_src

**Review by keur — APPROVED** [review]
views code LGTM

**Review by karenkliu — APPROVED** [review]
lgtm 👍

File: browser/ui/views/crash_report_permission_ask_dialog_view.cc [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  return false;
+}
+
+void CrashReportPermissionAskDialogView::OnDialogInitialized() {
+  SetButtonRowInsets(gfx::Insets(0, 0, 18, 24));
+}
+
+void CrashReportPermissionAskDialogView::OnAcceptButtonClicked() {
+  // Enable crash reporting.
+  ChangeMetricsReportingState(true);
```
Comment (simonhong) [inline browser/ui/views/crash_report_permission_ask_dialog_view.cc:253]: @iefremov can you check this? to enable crash reporting, calling this is sufficient?

**Comment by simonhong** [comment]
Hmm, new browser test cases causes timeout on macOS CI. checking..

**Pull request merged**